src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,268 @@
+        # updater. I hope they will work for many years.
+        return "https://dl.google.com/android/repository/" + (
+            "sdk-tools-{os}-4333796.zip".format(os=self.host_os.lower())
+        )
+
+    def verify_python_version(self):
+        if self.python_version_tag != "3.7":
+            raise BriefcaseCommandError(
+                """\
+Found Python version {self.python_version_tag}. Android packaging currently
+requires Python 3.7.""".format(
+                    self=self
+                )
+            )
+
+    def verify_sdk(self):
+        # If the SDK tools don't exist or aren't executable, install the SDK.
+        # TODO: Reconsider this for Windows.
+        tools_path = self.sdk_path / "tools" / "bin"
+        tools_ok = (
+            tools_path.exists()
+            and all(
+                [
+                    S_IMODE(tool.stat().st_mode) == 0o755
+                    for tool in tools_path.glob("*")
+                ]
+            )
+            and (tools_path / "sdkmanager").exists()
+        )
+        if tools_ok:
+            return
+
+        print("Setting up Android SDK...")
+        try:
+            sdk_zip_path = self.download_url(
+                url=self.sdk_url,
+                download_path=self.dot_briefcase / "tools",
+            )
+        except requests_exceptions.ConnectionError:
+            raise NetworkFailure("download Android SDK")
+        try:
+            with ZipFile(str(sdk_zip_path)) as sdk_zip:
+                sdk_zip.extractall(path=str(self.sdk_path))
+        except BadZipFile:
+            raise BriefcaseCommandError(
+                """\
+Invalid ZIP file found at {sdk_zip_path}
+
+Partial download? Remove it, then try again.""".format(
+                    sdk_zip_path=sdk_zip_path
+                )
+            )
+        sdk_zip_path.unlink()  # Zip file no longer needed once unpacked.
+        for binpath in tools_path.glob("*"):
+            binpath.chmod(0o755)
+
+    def verify_license(self):
+        license_path = self.sdk_path / "licenses" / "android-sdk-license"
+        if license_path.exists():
+            return
+
+        print("\nPlease accept `android-sdk-license` when prompted.\n")
+        try:
+            self.subprocess.run(
+                ["./sdkmanager", "--licenses"],
+                check=True,
+                cwd=str(self.sdk_path / "tools" / "bin"),
+            )
+        except subprocess.CalledProcessError:
+            raise BriefcaseCommandError(
+                """\
+Error while reviewing Android SDK licenses. Please run this command and examine
+its output for errors.
+
+$ {sdkmanager} --licenses""".format(
+                    sdkmanager=self.sdk_path / "tools" / "bin" / "sdkmanager"
+                )
+            )
+
+        if not license_path.exists():
+            raise BriefcaseCommandError(
+                """\
+You did not accept the Android SDK license. Please re-run the briefcase command
```

Comment:
`self.command` will be the command that was executed, so this can be made more context sensitive (i..e, "rerun 'briefcase run'...)

Reply:
I'm -0 on naming the specific command here, per Slack, because then we end up giving them information that might be possibly wrong for their environment (perhaps they're using briefcase as a library, or who knows what).

I'll keep this the way it is, or very similar, for now, but I'm OK with more pushback from you if you feel strongly!